adapters/sigfox-vertx/src/main/java/org/eclipse/hono/adapter/sigfox/impl/Config.java

```diff
@@ -0,0 +1,150 @@
+    private static final String BEAN_NAME_SIGFOX_PROTOCOL_ADAPTER = "sigfoxProtocolAdapter";
+    private static final String METRICS_PROTOCOL_TAG = "sigfox";
+
+    @PostConstruct
+    void validateConfiguration() {
+        final HttpProtocolAdapterProperties httpProtocolAdapterProperties = adapterProperties();
+        if (!httpProtocolAdapterProperties.isAuthenticationRequired()) {
+            throw new IllegalStateException(
+                    "LoRa Protocol Adapter does not support unauthenticated mode. Please change your configuration accordingly.");
+        }
+    }
+
+    /**
+     * Creates a new LoRa adapter instance.
+     * 
+     * @return The new instance.
+     */
+    @Bean(name = BEAN_NAME_SIGFOX_PROTOCOL_ADAPTER)
+    @Scope("prototype")
+    public SigfoxProtocolAdapter sigfoxProtocolAdapter() {
+        return new SigfoxProtocolAdapter();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void customizeDownstreamSenderFactoryConfig(final ClientConfigProperties props) {
+        if (props.getName() == null) {
+            props.setName(CONTAINER_ID_HONO_SIGFOX_ADAPTER);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void customizeRegistrationClientFactoryConfig(final RequestResponseClientConfigProperties props) {
+        if (props.getName() == null) {
+            props.setName(CONTAINER_ID_HONO_SIGFOX_ADAPTER);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void customizeCredentialsClientFactoryConfig(final RequestResponseClientConfigProperties props) {
+        if (props.getName() == null) {
+            props.setName(CONTAINER_ID_HONO_SIGFOX_ADAPTER);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void customizeTenantClientFactoryConfig(final RequestResponseClientConfigProperties props) {
+        if (props.getName() == null) {
+            props.setName(CONTAINER_ID_HONO_SIGFOX_ADAPTER);
+        }
+    }
+
+    /**
+     * Exposes the LoRa adapter's configuration properties as a Spring bean.
+     *
+     * @return The configuration properties.
+     */
+    @Bean
+    @ConfigurationProperties(prefix = "hono.http")
+    public HttpProtocolAdapterProperties adapterProperties() {
+        return new HttpProtocolAdapterProperties();
+    }
+
+    /**
+     * Exposes a factory for creating LoRa adapter instances.
+     *
+     * @return The factory bean.
+     */
+    @Bean
+    public ObjectFactoryCreatingFactoryBean serviceFactory() {
+        final ObjectFactoryCreatingFactoryBean factory = new ObjectFactoryCreatingFactoryBean();
+        factory.setTargetBeanName(BEAN_NAME_SIGFOX_PROTOCOL_ADAPTER);
+        return factory;
+    }
+
+    /**
+     * Gets the cache manager.
+     *
+     * @return the cache manager
+     */
+    @Bean
+    public CacheManager cacheManager() {
```

Comment:
I do not see where this is used. Can you try to remove it (along with the `@EnableCaching` annotation at the class level) and see if it has any impact? FMPOV it shouldn't ...

Reply:
Done. Indeed it doesn't.